Services/BookingService/BookingService.cs

```diff
@@ -0,0 +1,200 @@
+                ClinicId = dentalClinicInfo.ClinicId,
+                CustomerId = customerInfo.CustomerId,
+                DentistId = dentistInfo.StaffId,
+                CreationDate = DateTime.Now,
+                ScheduleSlotId = slotInfo.ScheduleSlotId,
+                Status = false
+            };
+
+            _unitOfWork._context.Bookings.Add(newBook);
+
+            message = "Suceed";
+            return true;
+        }
+
+        public bool CreateOneTimeAppointment(BookingRegistrationModel bookInfo, out string message)
+        {
+            var customerInfo = _unitOfWork._context.Customers.Include(x => x.User).Where(x => x.CustomerId == bookInfo.CustomerId).FirstOrDefault();
+
+            var dentistInfo = _unitOfWork._context.ClinicStaffs.Include(x => x.User).Where(x => x.StaffId == bookInfo.DentistId).FirstOrDefault();
+
+            var dentalClinicInfo = _unitOfWork._context.Clinics.Where(x => x.ClinicId == bookInfo.ClinicId).FirstOrDefault();
+
+            var slotInfo = _unitOfWork._context.ScheduledSlots.Include(x => x.Slot).Where(x => x.ScheduleSlotId == bookInfo.TimeSlotId && x.ClinicId == bookInfo.ClinicId).FirstOrDefault();
+
+            if (customerInfo == null)
+            {
+                message = "Customer information not found!";
+                return false;
+            }
+
+            if (dentistInfo == null)
+            {
+                message = "Dentist information not found!";
+                return false;
+            }
+
+            if (dentalClinicInfo == null)
+            {
+                message = "Clinic information not found!";
+                return false;
+            }
+
+            if (slotInfo == null)
+            {
+                message = "Slot information not found!";
+                return false;
+            }
+
+            var slotBookCount = _unitOfWork._context.Bookings.Where(x => x.DentistId == bookInfo.DentistId && x.AppointmentDate == bookInfo.AppointmentDate && x.ScheduleSlotId == bookInfo.TimeSlotId).Count();
+
+            if (slotInfo.MaxAppointments <= slotBookCount)
+            {
+                message = "The current slot is fully booked!";
+                return false;
+            }
+
+            var newBook = new Booking()
+            {
+                BookingType = "Khám t?ng quát",
+                AppointmentDate = bookInfo.AppointmentDate,
+                ClinicId = dentalClinicInfo.ClinicId,
+                CustomerId = customerInfo.CustomerId,
+                DentistId = dentistInfo.StaffId,
+                CreationDate = DateTime.Now,
+                ScheduleSlotId = slotInfo.ScheduleSlotId,
+                Status = false
+            };
+
+            _unitOfWork._context.Bookings.Add(newBook);
+
+            message = "Suceed";
+            return true;
+        }
+
+        public bool CreateSimpleAppointment(BookingModel bookInfo, IEnumerable<ClinicService> services, out string message)
```

Comment:
I will remove this later since we only need the BookingRegistrationModel for appointment booking creation.